### PR TITLE
Set expiry

### PR DESCRIPTION
**Pull request by xiafu-msft**

### CODE REVIEW

**Comment by xiafu-msft** [comment]
@rakshith91 This is the previous pr you commented https://github.com/Azure/azure-sdk-for-python/pull/11629

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_models.py [hunk 1]
```diff
@@ -326,10 +312,10 @@ class LeaseProperties(BlobLeaseProperties):
     :ivar str duration:
         When a file is leased, specifies whether the lease is of infinite or fixed duration.
     """
-    def __init__(self):
-        self.status = None
-        self.state = None
-        self.duration = None
+    def __init__(self, **kwargs):
+        super(LeaseProperties, self).__init__(
```
Comment (rakshith91) [inline sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_models.py:316]: how come this is not throwing a lint error? it's "useless super delegation"

File: sdk/storage/azure-storage-file-datalake/tests/test_file_async.py [hunk 2]
```diff
@@ -676,6 +676,28 @@ def test_get_properties_async(self):
...
+        file_client = await directory_client.create_file("newfile", metadata=metadata, content_settings=content_settings)
+        await file_client.set_file_expiry("Absolute", expires_on=expires_on)
+        properties = await file_client.get_file_properties()
+
+        # Assert
+        self.assertTrue(properties)
+        self.assertIsNotNone(properties.expiry_time)
+
+    @record
+    def test_set_expiry_async(self):
```
Comment (rakshith91) [inline sdk/storage/azure-storage-file-datalake/tests/test_file_async.py:697]: is there a specific timeline for migrating datalake tests to the devtools framewrok?

**Review by rakshith91 — DISMISSED** [review]
LGTM overall - just a couple questions

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_file_client.py [hunk 3]
```diff
@@ -246,8 +247,31 @@ def get_file_properties(self, **kwargs):
...
+            If the type of expires_on is datetime, it should be in UTC time.
+        :keyword int timeout:
+            The timeout parameter is expressed in seconds.
+        :rtype: None
+        """
+        try:
+            expires_on = convert_datetime_to_rfc1123(expires_on)
+        except AttributeError:
+            expires_on = str(expires_on)
+        return self._datalake_client_for_blob_operation.path\
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_file_client.py:273]: remove return

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/aio/_data_lake_file_client_async.py [hunk 4]
```diff
@@ -207,8 +208,30 @@ async def get_file_properties(self, **kwargs):
...
+            When expiry_options is RelativeTo*, expires_on should be an int in milliseconds
+        :keyword int timeout:
+            The timeout parameter is expressed in seconds.
+        :rtype: None
+        """
+        try:
+            expires_on = convert_datetime_to_rfc1123(expires_on)
+        except AttributeError:
+            expires_on = str(expires_on)
+        return await self._datalake_client_for_blob_operation.path.set_expiry(expiry_options, expires_on=expires_on,
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/aio/_data_lake_file_client_async.py:233]: remove return

**Review by tasherif-msft — APPROVED** [review]
LGTM :)

**Pull request merged**